okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -414,9 +445,12 @@ public void shutdown() {
       }
     }
 
-    // Send GOAWAY with lastGoodStreamId of 0, since we don't expect any server-initiated streams.
-    // The GOAWAY is part of graceful shutdown.
-    frameWriter.goAway(0, ErrorCode.NO_ERROR, new byte[0]);
+    // If the transport was never started, shutdown should not cause an error.
```

Comment:
Unless we have a reason to allow calling shutdown before start, I prefer prohibiting shutdown before start, at least stricter limit is better for exposing bugs.